pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -106,12 +107,12 @@ def __init__(
         self,
         model: models.Model | models.KnownModelName | None = None,
         *,
-        result_type: type[ResultData] = str,
         system_prompt: str | Sequence[str] = (),
         deps_type: type[AgentDeps] = NoneType,
         name: str | None = None,
         model_settings: ModelSettings | None = None,
         retries: int = 1,
+        result_type: type[ResultData] = str,
```

Comment:
I think this needs a different TypeVar (+overloads)? Its not the same type as Agent´s generic TResult, its basically a type override.

Reply:
See https://github.com/pydantic/pydantic-ai/issues/564#issuecomment-2567654787